admin-dev/themes/new-theme/js/pages/product/edit/product-form-model.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ObjectFormMapper from '@components/form/form-object-mapper';
+import ProductFormMapping from '@pages/product/edit/product-form-mapping';
+import ProductEventMap from '@pages/product/product-event-map';
+
+export default class ProductFormModel {
+  constructor($form, eventEmitter) {
+    this.mapper = new ObjectFormMapper(
+      $form,
+      ProductFormMapping,
+      eventEmitter,
+      {
+        modelUpdated: ProductEventMap.productModelUpdated,
+        updateModel: ProductEventMap.updatedProductModel,
+      },
+    );
+  }
+
+  getProduct() {
```

Comment:
@eternoendless the way I see it if we need a more strict model this is where it would be implemented We can have specific getters/setters that use the dynamic `ObjectFormMapper` as a core

Reply:
And for now it only handles product fields, but I guess we could add accessors to the combinations later, they just would be bound to another js component and this class would centralize it all